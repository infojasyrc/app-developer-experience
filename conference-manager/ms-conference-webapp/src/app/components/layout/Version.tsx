'use client';

import React from 'react';
import packageJson from '../../../../package.json';

const Version: React.FC = () => {
  return (
    <div className="fixed right-10 bottom-0 p-2 bg-white">
      Version: {packageJson.version}
    </div>
  );
};

export default Version;
