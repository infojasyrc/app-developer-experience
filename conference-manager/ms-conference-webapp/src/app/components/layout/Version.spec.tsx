import React from "react";
import { render, screen, waitFor, act } from "@testing-library/react";
import { describe, it, vi, expect, beforeEach } from "vitest";

import Version from "./Version";

vi.mock("../../../../package.json", () => ({
  default: {
    version: "2.0.0",
  },
}));

describe("Version component", () => {
  beforeEach(() => {
    // Reset mocks before each test
    vi.resetModules();
  });

  it("should render the version from the mocked package.json", async () => {
    render(<Version />);
    const version = "Version: 2.0.0";
    const txtVersion = screen.getByText(version);
    expect(txtVersion).toBeInTheDocument();
  });

  it("should match snapshot with mocked version", async () => {
    const { asFragment } = render(<Version />);
    expect(asFragment()).toMatchSnapshot();
  });
});
