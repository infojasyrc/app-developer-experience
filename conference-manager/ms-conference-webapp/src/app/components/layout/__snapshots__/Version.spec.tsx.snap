// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Version component > should match snapshot with mocked version 1`] = `
<DocumentFragment>
  <div
    class="fixed right-10 bottom-0 p-2 bg-white"
  >
    Version: 2.0.0
  </div>
</DocumentFragment>
`;
